File: java/src/Camera.java
public class Camera implements IPositionnable, IAllumable {

    private boolean allumer = false;
    private float x = 0;
    private float y = 0;
    private float z = 0;


    @Override
    public void allumer() {
        this.allumer = true;
    }

    @Override
    public void eteindre() {
        this.allumer = false;
    }


    @Override
    public float getX() {
        return this.x;
    }

    @Override
    public float getY() {
        return this.y;
    }

    @Override
    public float getZ() {
        return this.z;
    }

    @Override
    public void setX(float x) {
        this.x = x;
    }

    @Override
    public void setY(float y) {
        this.y = y;
    }

    @Override
    public void setZ(float z) {
        this.z = z;
    }

    @Override
    public void deplacer(float x, float y, float z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }
}

File: csharp/Camera.cs
/*
 * Created by SharpDevelop.
 * User: GameBase
 * Date: 25/02/2018
 * Time: 17:43
 * 
 * To change this template use Tools | Options | Coding | Edit Standard Headers.
 */
using System;

namespace Dpo_project
{
	/// <summary>
	/// Description of Camera.
	/// </summary>
	public class Camera : IAllumable, IPositionnable
	{
		private float x;
		private float y;
		private float z;
		
		public Camera()
		{
		}
		
		public void deplacer() {}
		public void allumer() {}
		public void eteindre() {}
		
		public void calculerSurface() {}
		public void set() {}
		public void setAttr(float x, float y, float z) {
			this.x = x;
			this.y = y;
			this.z = z;
		}
		public float getX() {
			return x;
		}
		public float getY() {			
			return y;
		}
		public float getZ() {
			return z;
		}
	}
}
